src/main/scala/sample/reactivekafka/KafkaReaderCoordinator.scala

```diff
@@ -41,19 +41,53 @@ class KafkaReaderCoordinator(mat: Materializer, topicName: String) extends Actor
       .kafkaOffsetsStorage()
       .commitInterval(1200 milliseconds)
     )
+
     log.debug("Starting the reader")
-    Source(consumerWithOffsetSink.publisher)
-      .map(processMessage)
+    val currencyRateSource = Source(consumerWithOffsetSink.publisher)
+    val currencyBroadcaster = Source(currencyRateSource.runWith(Sink.publisher(true)))
+
+    val alertsFlow = Flow[MessageAndMetadata[Array[Byte], CurrencyRateUpdated]]
+      .map(processAlert)
+      .filter(alert => alert != None)
+      .map(alert => alert.get)
+
+
+    // commit offset so we don't process the kafka message again
+    currencyBroadcaster
       .withAttributes(ActorAttributes.supervisionStrategy(processingDecider))
-      .to(consumerWithOffsetSink.offsetCommitSink).run()
+      .to(consumerWithOffsetSink.offsetCommitSink)
```

Comment:
I'm writing three streams explicitly to show that these are three explicit streams.
Your way of writing the flow uses the DSL to describe what the flow is.

For me this is a big turn off because
* DSL adds another layer of indirection which I have to learn
* the flow has to be specified beforehand, which is not plausible for non-closed flows
* I wanted an unbounded number of sinks to be able to subscribe to the publisher

From a beginner's point of view I view separating these into three separate streams to be more readable and more comprehensible than the scala DSL you have provided.